parser/inline.go

```diff
@@ -766,7 +766,18 @@ func entity(p *Parser, data []byte, offset int) (int, ast.Node) {
 	// undo &amp; escaping or it will be converted to &amp;amp; by another
 	// escaper in the renderer
 	if bytes.Equal(ent, []byte("&amp;")) {
-		ent = []byte{'&'}
+		return end, newTextNode([]byte{'&'})
+	}
+	// if ent consists solely out of numbers (hex or decimal) convert that unicode codepoint to actual rune
+	codepoint := uint64(0)
+	var err error
+	if ent[2] == 'x' || ent[2] == 'X' { // hexadecimal
```

Comment:
Let's add test for things like `&;`, `&#;`, `&#1;`, `&#x3;` to make sure we don't crash with out-of-bounds. I can't tell just from reading code that e.g. `x[2]` is always safe.